### DISCUSSION TITLE
Velocity unsteady

### DISCUSSION THREAD

**Discussion by MaikFZJ** [PedestrianDynamics/PedPy#195] [General]
compute_individual_velocity calculates unsteady velocity depending on frame_step at the beginning and end of the trajectory. The interval does not increasing gradually and has the current frame in the centre.

**Comment by schroedtert** [comment]
After discussion:
- leave current version in
- add new velocity methods, switch with parameter
  - exclude parts where not the whole frame interval is valid
  - use an adaptive approach (increase frame interval till it reaches full width)

**Comment by schroedtert** [comment]
Differences with some newly implemented functions:

- current/single sided: use only one side, if on the other less than frame_step frames are available
- cutting/excluded: use only frames where both sides are complete
- adaptive: use a balanced window, with as many frames available
- mixed/maximum range: use as many frames as possible


![image](https://github.com/PedestrianDynamics/PedPy/assets/15690524/bf00bb84-0b43-4791-bf41-96428a9d3fd8)

Results (without cutting, as no difference in the middle):

![image](https://github.com/PedestrianDynamics/PedPy/assets/15690524/d13dd583-6338-4538-8f51-e577b083912a)

**Comment by schroedtert** [comment]
Added the above discussed methods:
- exclude border
- adaptive border
- single sided border

More detailed description here: https://pedpy.readthedocs.io/en/stable/user_guide.html#individual-speed
